07-generic_bookworms/final/collectors/collector.go

```diff
@@ -0,0 +1,75 @@
+package collectors
+
+import (
+	"fmt"
+	"io"
+	"sort"
+)
+
+// A Collector contains the list of books on a collector's shelf.
+type Collector[T collectible] struct {
+	Name  string `json:"name"`
+	Items []T    `json:"items"`
+}
+
+type collectible interface {
+	comparable
+	Sortable
+}
+
+// Sortable exposes a way of telling if an item should appear before another.
+// Sortable needs to be exposed so that we can use Before elsewhere.
+type Sortable interface {
+	Before(Sortable) bool
+}
+
+// Collectors is a list of collectors.
+type Collectors[T collectible] []Collector[T]
+
+// FindCommon returns items that are on more than one collector's shelf.
+func (c Collectors[T]) FindCommon() []T {
```

Comment:
I'm not quite following the purpose of this func. Given enough people this would return, pretty much, every single book. So, it doesn't seem that useful. Maybe there is a better example.